File: dotnet-adapters-shortest-paths-core/test/java/com/programmerare/shortestpaths/utils/MapperForIntegerIdsAndGeneralStringIdsTest.java
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
package com.programmerare.shortestpaths.utils;

import static org.junit.Assert.*;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * @author Tomas Johansson
 */
public class MapperForIntegerIdsAndGeneralStringIdsTest {

//	@Before
//	public void setUp() throws Exception {
//	}

	// TODO: improve the testing below. Each method is currently doing too much.

	@Test
	public void testStartIndexZero() {
		MapperForIntegerIdsAndGeneralStringIds idMapper = MapperForIntegerIdsAndGeneralStringIds.createIdMapper(0);
		assertEquals(0, idMapper.createOrRetrieveIntegerId("A"));
		assertEquals(1, idMapper.createOrRetrieveIntegerId("B"));
		assertEquals(0, idMapper.createOrRetrieveIntegerId("A"));
		assertEquals(2, idMapper.createOrRetrieveIntegerId("C"));
		assertEquals(0, idMapper.createOrRetrieveIntegerId("A"));
		assertEquals(3, idMapper.createOrRetrieveIntegerId("D"));

		assertEquals(4, idMapper.getNumberOfVertices());
		
		assertEquals("A", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(0));
		assertEquals("B", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(1));
		assertEquals("C", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(2));
		assertEquals("D", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(3));
	}
	
	@Test
	public void testStartIndexOne() {
		MapperForIntegerIdsAndGeneralStringIds idMapper = MapperForIntegerIdsAndGeneralStringIds.createIdMapper(1);
		assertEquals(1, idMapper.createOrRetrieveIntegerId("ABC"));
		assertEquals(2, idMapper.createOrRetrieveIntegerId("DEF"));
		assertEquals(3, idMapper.createOrRetrieveIntegerId("GHI"));
		assertEquals(2, idMapper.createOrRetrieveIntegerId("DEF"));
		
		assertEquals(3, idMapper.getNumberOfVertices());
		
		assertEquals("ABC", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(1));
		assertEquals("DEF", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(2));
		assertEquals("GHI", idMapper.getBackThePreviouslyStoredGeneralStringIdForInteger(3));
	}	
	
//	@Test
//	public void createOrRetrieveIntegerId() {
//		fail("Not yet implemented");
//	}
//
//	@Test
//	public void testGetNumberOfVertices() {
//		fail("Not yet implemented");
//	}
//
//	@Test
//	public void testGetBackThePreviouslyStoredGeneralStringIdForInteger() {
//		fail("Not yet implemented");
//	}
}

File: dotnet-adapters-shortest-paths-test/test/com.programmerare.shortestpaths/utils/MapperForIntegerIdsAndGeneralStringIdsTest.cs
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code in the "core" project is licensed with MIT.
* Other projects within this Visual Studio solution may be released with other licenses e.g. Apache.
* Please find more information in the files "License.txt" and "NOTICE.txt" 
* in the project root directory and/or in the solution root directory.
* It should also be possible to find more license information at this URL:
* https://github.com/TomasJohansson/adapters-shortest-paths-dotnet/
*/

using NUnit.Framework;
using static NUnit.Framework.Assert;

namespace com.programmerare.shortestpaths.utils
{
[TestFixture]
public class MapperForIntegerIdsAndGeneralStringIdsTest {

//	@Before
//	public void setUp() throws Exception {
//	}

	// TODO: improve the testing below. Each method is currently doing too much.

	[Test]
	public void testStartIndexZero() {
		MapperForIntegerIdsAndGeneralStringIds idMapper = MapperForIntegerIdsAndGeneralStringIds.CreateIdMapper(0);
		AreEqual(0, idMapper.CreateOrRetrieveIntegerId("A"));
		AreEqual(1, idMapper.CreateOrRetrieveIntegerId("B"));
		AreEqual(0, idMapper.CreateOrRetrieveIntegerId("A"));
		AreEqual(2, idMapper.CreateOrRetrieveIntegerId("C"));
		AreEqual(0, idMapper.CreateOrRetrieveIntegerId("A"));
		AreEqual(3, idMapper.CreateOrRetrieveIntegerId("D"));

		AreEqual(4, idMapper.GetNumberOfVertices());
		
		AreEqual("A", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(0));
		AreEqual("B", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(1));
		AreEqual("C", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(2));
		AreEqual("D", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(3));
	}
	
	[Test]
	public void testStartIndexOne() {
		MapperForIntegerIdsAndGeneralStringIds idMapper = MapperForIntegerIdsAndGeneralStringIds.CreateIdMapper(1);
		AreEqual(1, idMapper.CreateOrRetrieveIntegerId("ABC"));
		AreEqual(2, idMapper.CreateOrRetrieveIntegerId("DEF"));
		AreEqual(3, idMapper.CreateOrRetrieveIntegerId("GHI"));
		AreEqual(2, idMapper.CreateOrRetrieveIntegerId("DEF"));
		
		AreEqual(3, idMapper.GetNumberOfVertices());
		
		AreEqual("ABC", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(1));
		AreEqual("DEF", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(2));
		AreEqual("GHI", idMapper.GetBackThePreviouslyStoredGeneralStringIdForInteger(3));
	}	
	
//	@Test
//	public void createOrRetrieveIntegerId() {
//		fail("Not yet implemented");
//	}
//
//	@Test
//	public void testGetNumberOfVertices() {
//		fail("Not yet implemented");
//	}
//
//	@Test
//	public void testGetBackThePreviouslyStoredGeneralStringIdForInteger() {
//		fail("Not yet implemented");
//	}
}
}